components/brave_shields/browser/tracking_protection_service.cc

```diff
@@ -10,36 +10,210 @@
+                                                     int new_render_process_id,
+                                                     int new_render_frame_id) {
+  DCHECK_CURRENTLY_ON(BrowserThread::IO);
+  const RenderFrameIdKey old_key(old_render_process_id, old_render_frame_id);
+  auto iter = render_frame_key_to_starting_site_url.find(old_key);
+  if (iter != render_frame_key_to_starting_site_url.end()) {
+    const RenderFrameIdKey new_key(new_render_process_id, new_render_frame_id);
+    render_frame_key_to_starting_site_url.insert(
+        std::pair<RenderFrameIdKey, GURL>(new_key, iter->second));
+    render_frame_key_to_starting_site_url.erase(old_key);
+  }
+}
+
+void TrackingProtectionService::DeleteRenderFrameKey(int render_process_id,
+                                                     int render_frame_id) {
+  DCHECK_CURRENTLY_ON(BrowserThread::IO);
+  const RenderFrameIdKey key(render_process_id, render_frame_id);
+  render_frame_key_to_starting_site_url.erase(key);
+}
+
+bool TrackingProtectionService::ShouldStoreState(HostContentSettingsMap* map,
+                                                 int render_process_id,
+                                                 int render_frame_id,
+                                                 const GURL& top_origin_url,
+                                                 const GURL& origin_url) const {
+  DCHECK_CURRENTLY_ON(BrowserThread::IO);
+  if (!TrackingProtectionHelper::IsSmartTrackingProtectionEnabled()) {
+    return true;
+  }
+
+  if (first_party_storage_trackers_.empty()) {
+    LOG(INFO) << "First party storage trackers list is empty";
+    return true;
+  }
+
+  const std::string host = origin_url.host();
+  const GURL starting_site =
+      GetStartingSiteForRenderFrame(render_process_id, render_frame_id);
+
+  // If starting host is the current host, user-interaction has happened
+  // so we allow storage
+  if (starting_site.host() == host) {
+    return true;
+  }
+
+  const bool allow_brave_shields =
+      starting_site.is_empty()
+          ? false
+          : IsAllowContentSetting(map, starting_site, GURL(),
+                                  CONTENT_SETTINGS_TYPE_PLUGINS,
+                                  brave_shields::kBraveShields);
+
+  if (!allow_brave_shields) {
+    return true;
+  }
+
+  const bool allow_trackers =
+      starting_site.is_empty()
+          ? true
+          : IsAllowContentSetting(map, starting_site, GURL(),
+                                  CONTENT_SETTINGS_TYPE_PLUGINS,
+                                  brave_shields::kTrackers);
+
+  if (allow_trackers) {
+    return true;
+  }
+
+  // deny storage if host is found in the tracker list
+  return first_party_storage_trackers_.find(host) ==
+         first_party_storage_trackers_.end();
+}
+
+void TrackingProtectionService::ParseStorageTrackersData() {
+  if (storage_trackers_buffer_.empty()) {
+    LOG(ERROR) << "Could not obtain tracking protection data";
```

Comment:
is this an error? If not it should be `INFO` and should we do something more than log?

Reply:
@bridiver - I used the same semantics as `OnDATFileReady`:

https://github.com/brave/brave-core/blob/7c13b83f5aecc7628c8fce7e81cbdadbd06a88fa/components/brave_shields/browser/tracking_protection_service.cc#L77

I don't think we should update it to `INFO`, it highlights a corrupted `TrackingList`. Not sure if we can do something additional here.